### DISCUSSION TITLE
Machine Learning Framework

### DISCUSSION THREAD

**Discussion by ananyas168** [petermr/petermr#19] [General]
This channel is to discuss the course of actions, we are going to take to successfully build A classification model for the terpene classification task.

**Comment by ananyas168** [comment]
# Ideas on Terpene Classification
The classification of the terpene pathway images can be done in via two different paths:
1.  Supervised Learning
2. Unsupervised Learning

Below the figures depicts the estimated framework for the supervised and unsupervised method:

###  Supervised Learning
![supervised_terpene_classification](https://user-images.githubusercontent.com/66965350/154963154-f174200e-41d9-4ca7-93dc-6c99e6f268b7.png)


### Unsupervised Learning
![Unsupervised Learning](https://github.com/petermr/CEVOpen/blob/master/statistical_ananlysis_2022/example_data/unsupervised_terpene_classification.png)
![unsupervised_terpene_classification](https://user-images.githubusercontent.com/66965350/154961909-867765ed-188d-4507-b80f-d0309035fab1.png)
Step to be followed:
1. Obtain the data from images using Tesseract
2. Clean them as far as possible

3. create training and testing data
4. Select an NLP model to compute embeddings( scispacy, Bio-Bert, etc)
5. Select a Binary classifier
6. Train the model and analyze performance.

**Comment by petermr** [comment]
Thank you!


On Mon, Feb 21, 2022 at 8:00 AM Ananya ***@***.***> wrote:

> This channel is to discuss the course of actions, we are going to take to
> successfully build A classification model for the terpene classification
> task.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/petermr/petermr/discussions/19>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFTCS4VEP3PFQ5APAUCYWDU4HWJRANCNFSM5O5ZWSFA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Peter Murray-Rust
Founder ContentMine.org
and
Reader Emeritus in Molecular Informatics
Dept. Of Chemistry, University of Cambridge, CB2 1EW, UK

**Comment by ananyas168** [comment]
Updates on my work on step no 4 - Select an NLP model to compute embeddings

I have tested scispacy for embedding generation and it performs quite well.
Still need some time to fix the BioBert code there is some package issues. But I will surely solve it.:)

**Comment by ananyas168** [comment]
**TODO for caption classification:**
1. Get the pmcid from alexpico in bulk.
2. Run pygetpapers to get the paper related to the pmcid in XML format
3.  Run docanalysis to get figure captions from the papers
4. Use scispacy to do chemical entity identification( building the training dataset)
5. Run logistic regression on it to see how it performs. Before that we will be using a nlp model to compute embeddings.

**TODO for image classification:**
1. get the words from the images( output from teserract)
2. generate training data
3. use entity identifier to help you understand and labeling
4. use biobert to compute embedding.
5. Train basic classifier (logistic regression) for classification
6. look into te results 
7. think about modification
8. need to run a few loops over the process to get the best performance.
